frontend/controller/service-worker.js

```diff
@@ -204,11 +216,17 @@ sbp('sbp/selectors/register', {
         let subID = null
         // get a real push subscription only if both browser permissions allow and user wants us to
         if (notificationEnabled && granted) {
+          const subscriptionOptions = await sbp('push/getSubscriptionOptions')
           subscription = await registration.pushManager.getSubscription()
+          if (subscription && !bufferEq(subscription.options.applicationServerKey, subscriptionOptions?.applicationServerKey)) {
+            console.warn('VAPID server key changed; removing existing subscription and setting up a new one', { oldKey: subscription.options.applicationServerKey, newKey: subscriptionOptions.applicationServerKey })
```

Comment:
Should you be logging these keys (especially the new key)? If so, can you add a comment explaining why it's safe to do so? Remember users could be sending us these logs and we don't want to expose any secrets.

Reply:
Well, these keys are public information (or at least, public keys) and they belong to the server, so I don't see this being an issue.